cpp/notebooks/jupyter_ros_utils/ros2_jupyter.hpp

```diff
@@ -160,39 +285,44 @@ class JupyterNodeAdvanced: ROS2Init, public NodeType {
         }
         ros_valid.value = rclcpp::ok();
     }
-    
+
     void setup_thread() {
         if (spin_thread.joinable()) {
             // should never happen
             spin_thread.join();
         }
-        spin_thread = std::thread([this]{ 
+        spin_thread = std::thread([this]{
             ros_valid.value = rclcpp::ok();
-            while(rclcpp::ok()) { 
-                rclcpp::spin_some(getNode()); 
+            while(rclcpp::ok()) {
+                rclcpp::spin_some(getNode());
                 m_rate.sleep();
             }
-            ros_valid.value = false; 
+            ros_valid.value = false;
         });
-    
     }
 
-    public:
+public:
     std::shared_ptr<NodeType> getNode() {
         return this->shared_from_this();
     }
 
-    JupyterNodeAdvanced(double rate = 10): NodeType("interative_node"), m_rate(rate) {
+    JupyterNodeAdvanced(double rate = 10): NodeType("interactive_node"), m_rate(rate) {
         spin_btn.description = "Stop";
         spin_btn.on_click([this]{toggle();});
-        
+
         disp_box.add(spin_btn);
         disp_box.add(ros_valid);
         disp_box.display();
-        
+
         setup_thread();
     }
 
+
+public:
+    auto create_param_widgets() {
```

Comment:
Can we follow the ROS2 format for this? Ad 2 functions:
* set_parameter
* get_parameter
They create the same widget, but in different states (get_param widget is already spinning, while set_param widget is waiting for user input)

Reply:
what is it supposed to do here? `JupyterNodeAdvanced` is already an Ros2 node, and already has `get_parameters` / `set_parameters`.

Reply:
Can we overload the functions since they are not virtual?

Reply:
Not possible directly to use the ros2 parameter code.
return types of `get_parameter` of `rcldpp::Node` is  `rclcpp::Parameter`
https://docs.ros2.org/bouncy/api/rclcpp/classrclcpp_1_1_node.html#a0bf1077fba623d72fe1b49805f6c0a5a
but `widget` also needs to be returned.